### ISSUE TITLE
[BUG]: Autoencoder Model Training fails, if Feature set contains only numerical features.

### ISSUE THREAD

**Issue by spatil44** [nv-morpheus/Morpheus#1423]
### Version

23.07

### Which installation method(s) does this occur on?

Docker, Kubernetes

### Describe the bug.

If the feature set comprises solely numerical features, the Autoencoder Model experiences failure during the Model training stage.

### Minimum reproducible example

```shell
from morpheus.models.dfencoder import AutoEncoder
import pandas as pd
from collections import OrderedDict

data = [[1, 10], [4, 15], [5, 14]]
num_df = pd.DataFrame(data, columns=['Name', 'Age'])

model = AutoEncoder(
        encoder_layers=[256],  # layers of the encoding part
        decoder_layers=[256],  # layers of the decoding part
        activation="relu",  # activation function
        swap_p=0.2,  # noise parameter
        lr=0.01,  # learning rate
        lr_decay=0.99,  # learning decay
        batch_size=256,
        logger="basic",
        verbose=True,
        progress_bar=True,
        optimizer="sgd",  # SGD optimizer is selected (Stochastic gradient descent)
        scaler="standard",  # feature scaling method
        min_cats=1,  # cutoff for minority categories
        device="cuda"
    )
model.fit(num_df, epochs=2)
```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

File /opt/conda/envs/morpheus/lib/python3.10/site-packages/morpheus/models/dfencoder/autoencoder.py:1555, in AutoEncoder.get_anomaly_score(self, df)
   1549 """
   1550 Returns a per-row loss of the input dataframe.
   1551 Does not corrupt inputs.
   1552 """
   1553 mse, bce, cce = self.get_anomaly_score_losses(df)
-> 1555 combined_loss = torch.cat([mse, bce, cce], dim=1)
   1557 net_loss = combined_loss.mean(dim=1).cpu().numpy()
   1559 return net_loss

RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument tensors in method wrapper_CUDA_cat)

</pre></details>

### Full env printout

<details><summary>Click here to see environment details</summary><pre>

NA

</pre></details>

### Other/Misc.

-
Sandip Patil | Nvidia

### Code of Conduct

- [X] I agree to follow Morpheus' Code of Conduct
- [X] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

**Comment by jarmak-nv** [comment]
Hi @spatil44!
  
  Thanks for submitting this issue - our team has been notified and we'll get back to you as soon as we can!
  In the mean time, feel free to add any relevant information to this issue.

**Comment by efajardo-nv** [comment]
@spatil44 The AutoEncoder code in your snippet indicates that you're using v23.07 (keyword arguments were updated in v23.11). I was able to reproduce your error in my v23.07 environment but not in v23.11. Could you also try with v23.11 to see if it resolves your issue? AutoEncoder constructor call would need to be updated to this:

```
model = AutoEncoder(
        encoder_layers=[256],  # layers of the encoding part
        decoder_layers=[256],  # layers of the decoding part
        activation="relu",  # activation function
        swap_probability=0.2,  # noise parameter
        learning_rate=0.01,  # learning rate
        learning_rate_decay=0.99,  # learning decay
        batch_size=256,
        logger="basic",
        verbose=True,
        progress_bar=True,
        optimizer="sgd",  # SGD optimizer is selected (Stochastic gradient descent)
        scaler="standard",  # feature scaling method
        min_cats=1,  # cutoff for minority categories
        device="cuda"
    )

**Comment by spatil44** [comment]
Thanks @efajardo-nv  for an update.  @hsin-c  tested on 23.11, she was also getting the same error.
However, I'll give try on 23.11 and will update here.

**Comment by spatil44** [comment]
The issue is solved on the 23.11 branch, so closing this.
